Firestore/Source/Core/FSTQuery.mm

```diff
@@ -214,37 +240,43 @@ - (BOOL)isEqual:(id)other {
   if (other == self) return YES;
   if (![[other class] isEqual:[self class]]) return NO;
 
-  return [self.field isEqual:((FSTNullFilter *)other).field];
+  return _field == ((FSTNullFilter *)other)->_field;
 }
 
 - (NSUInteger)hash {
-  return [self.field hash];
+  return _field.Hash();
```

Comment:
For future reference: `NSUInteger` and `size_t` aren't guaranteed to be the same size (though I think on arm32 and arm64 they happen to be).